source/tyr/tyr/resources.py

```diff
@@ -269,7 +87,7 @@ def post(self, instance_name):
         instance = models.Instance.query_existing().filter_by(name=instance_name).first_or_404()
 
         if not request.files:
-            return {'message': 'the Data file is missing'}, 400
+            return {'message': 'the Data file is missing'}, 404
```

Comment:
[correction of this test](https://github.com/CanalTP/navitia/blob/dev/source/tyr/tests/integration/datapost_test.py#L38)